### PR TITLE
SOV-625 remove governance withdraw tokens

### PR DESCRIPTION
**Pull request by cwsnt**

### CODE REVIEW

File: contracts/governance/Vesting/VestingLogic.sol [hunk 1]
```diff
@@ -140,7 +141,11 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
      * @param isGovernance Whether all tokens (true)
      * or just unlocked tokens (false).
      * */
-    function _withdrawTokens(address receiver, bool isGovernance) internal {
+    function _withdrawTokens(
```
Comment (tjcloa) [inline contracts/governance/Vesting/VestingLogic.sol:144]: this implementation doesn't solve the block gas limit issue - it should be refactored to allow partial withdrawals using [startFrom, maxIterations] range pattern 
because vesting contracts are not upgradable, i suggest adding these as parameters to the function call rather than adding to the storage because changing them will be tricky

**Review by tjcloa — CHANGES_REQUESTED** [review]

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 2]
```diff
@@ -352,9 +354,21 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
...
-        vestingWhitelist[vesting] = true;
-        ITeamVesting(vesting).governanceWithdrawTokens(receiver);
-        vestingWhitelist[vesting] = false;
+        require(vestingRegistryLogic.isTeamVesting(vesting), "Only team vesting allowed");
+
+        ITeamVesting teamVesting = ITeamVesting(vesting);
+        uint256 teamVestingStartDate = teamVesting.startDate();
+        uint256 teamVestingCliff = teamVesting.cliff();
+
+        uint256 startFrom = teamVestingStartDate + teamVestingCliff;
```
[the remark this answers is not in the document]
Comment (dharkmattr) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol:363]: This is a way to cancel a vesting contract. Should it be possible to cancel even before the cliff? Why does the withdrawal begin only after the cliff?

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 3]
```diff
@@ -99,7 +99,7 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
         address _vesting,
         address _receiver,
         uint256 _startFrom
-    ) private {
+    ) private returns (uint256 end, bool isPartiallyCancelled) {
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/00dd8d0062afe84664e64d8efc4b38abc4df7cb3)

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 4]
```diff
@@ -140,8 +140,10 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
 
         if (adjustedEnd < end) {
             emit TeamVestingPartiallyCancelled(msg.sender, _receiver, adjustedEnd);
+            return (adjustedEnd, true);
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/00dd8d0062afe84664e64d8efc4b38abc4df7cb3)

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 5]
```diff
@@ -352,9 +354,21 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
...
+        uint256 teamVestingCliff = teamVesting.cliff();
+
+        uint256 startFrom = teamVestingStartDate + teamVestingCliff;
+        bool withdrawFlag = true;
+
+        while (withdrawFlag) {
+            (uint256 end, bool isPartiallyCancelled) =
+                _cancelTeamVesting(vesting, receiver, startFrom);
+            startFrom = end + TWO_WEEKS;
+            withdrawFlag = isPartiallyCancelled ? true : false;
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/00dd8d0062afe84664e64d8efc4b38abc4df7cb3)

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 6]
```diff
@@ -352,9 +354,21 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
...
+
+        ITeamVesting teamVesting = ITeamVesting(vesting);
+        uint256 teamVestingStartDate = teamVesting.startDate();
+        uint256 teamVestingCliff = teamVesting.cliff();
+
+        uint256 startFrom = teamVestingStartDate + teamVestingCliff;
+        bool withdrawFlag = true;
+
+        while (withdrawFlag) {
+            (uint256 end, bool isPartiallyCancelled) =
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/00dd8d0062afe84664e64d8efc4b38abc4df7cb3)

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 7]
```diff
@@ -352,9 +354,21 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
...
+                _cancelTeamVesting(vesting, receiver, startFrom);
+            startFrom = end + TWO_WEEKS;
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/00dd8d0062afe84664e64d8efc4b38abc4df7cb3)

File: contracts/governance/Vesting/VestingLogic.sol [hunk 8]
```diff
@@ -140,7 +141,11 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
...
+        address receiver,
+        bool isGovernance,
+        uint256 endRegularWithdrawal
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Vesting/VestingLogic.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/c3979c0614ca2a85fd4e60aae785d2ff16cef7de)

File: contracts/governance/Vesting/VestingLogic.sol (hunk 1 above)
Comment (cwsnt) [reply contracts/governance/Vesting/VestingLogic.sol:154]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/c3979c0614ca2a85fd4e60aae785d2ff16cef7de)

File: contracts/governance/Vesting/VestingLogic.sol [hunk 9]
```diff
@@ -110,26 +110,35 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
...
+        _withdrawTokens(receiver, false, startFrom, block.timestamp);
     }
 
     /**
-     * @notice Withdraws unlocked tokens from the staking contract and
+     * @notice Withdraws unlocked tokens partially (based on the max withdraw iteration that has been set) from the staking contract and
      * forwards them to an address specified by the token owner.
      * @param receiver The receiving address.
+     * @param startFrom The start value for the iterations.
+     * @param maxWithdrawIterations max withdrawal iteration
```
Comment (tjcloa) [inline contracts/governance/Vesting/VestingLogic.sol:127]:
```suggestion
     * @param maxWithdrawIterations max withdrawal iteration to work around block gas limit issue
```

**Review by tjcloa — CHANGES_REQUESTED** [review]
see inline comments

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol:367]: In this case, the cancellation always calculates after the cliff, since the available token can be withdrawn is effective after the cliff.

**Review by dharkmattr — CHANGES_REQUESTED** [review]

File: contracts/governance/Vesting/VestingLogic.sol [hunk 10]
```diff
@@ -138,9 +147,16 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
      * @dev Once here the caller permission is taken for granted.
      * @param receiver The receiving address.
      * @param isGovernance Whether all tokens (true)
+     * @param startFrom start withdrawal from
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Vesting/VestingLogic.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/5d16b2b3332e1572904944c498d69c3c71bee06c)

File: contracts/governance/Vesting/VestingLogic.sol (hunk 9 above)
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Vesting/VestingLogic.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/5d16b2b3332e1572904944c498d69c3c71bee06c)

File: contracts/governance/Vesting/VestingLogic.sol [hunk 11]
```diff
@@ -110,26 +110,35 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
...
+        uint256 startFrom,
+        uint256 maxWithdrawIterations
+    ) public onlyOwners {
+        uint256 defaultStartFrom = startDate + cliff;
+
+        startFrom = startFrom < defaultStartFrom ? defaultStartFrom : startFrom;
+
+        // @dev max iterations need to be decreased by 1, otherwise the iteration will always be surplus by 1
+        uint256 totalIterationValue = (startFrom + (FOUR_WEEKS * (maxWithdrawIterations - 1)));
+        uint256 withdrawUntil = endDate < totalIterationValue ? endDate : totalIterationValue;
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Vesting/VestingLogic.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/5d16b2b3332e1572904944c498d69c3c71bee06c)

File: contracts/governance/Vesting/VestingLogic.sol [hunk 12]
```diff
@@ -138,9 +147,16 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
...
+     * @param startFrom start withdrawal from
+     * @param withdrawUntil end time for regular withdrawal
      * or just unlocked tokens (false).
      * */
-    function _withdrawTokens(address receiver, bool isGovernance) internal {
+    function _withdrawTokens(
+        address receiver,
+        bool isGovernance,
+        uint256 startFrom,
+        uint256 withdrawUntil
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Vesting/VestingLogic.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/5d16b2b3332e1572904944c498d69c3c71bee06c)

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 13]
```diff
@@ -94,12 +94,15 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
      * @param _receiver The receiving address.
      * @param _startFrom The start value for the iterations.
      * or just unlocked tokens (false).
+     *
+     * @return nextStartFrom the value of start from to be used for next withdrawal.
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/5d16b2b3332e1572904944c498d69c3c71bee06c)

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 14]
```diff
@@ -139,9 +142,12 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
         }
 
         if (adjustedEnd < end) {
+            nextStartFrom = adjustedEnd + TWO_WEEKS;
             emit TeamVestingPartiallyCancelled(msg.sender, _receiver, adjustedEnd);
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/5d16b2b3332e1572904944c498d69c3c71bee06c)

File: contracts/governance/Vesting/VestingLogic.sol [hunk 15]
```diff
@@ -110,26 +110,35 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
...
-    function withdrawTokens(address receiver) public onlyOwners {
-        _withdrawTokens(receiver, false);
+    function withdrawTokensStartingFrom(
+        address receiver,
+        uint256 startFrom,
+        uint256 maxWithdrawIterations
+    ) public onlyOwners {
+        uint256 defaultStartFrom = startDate + cliff;
+
+        startFrom = startFrom < defaultStartFrom ? defaultStartFrom : startFrom;
```
[the remark this answers is not in the document]
Comment (tjcloa) [reply contracts/governance/Vesting/VestingLogic.sol:136]: sorry, that must have been confusing   
lm clarify  
the original `_withdrawTokens` function used this cycle:  
`for (uint256 i = startDate + cliff; i <= end; i += FOUR_WEEKS) {`   
meaning that to keep consistent, startForm parameter in the updated function should be adjusted to `startDate + cliff + n * FOUR_WEEKS` - you can take this adjustment logic from the `timestampToLockDate` in Staking.   
if it is not, further calculations would be messed up.  
i also suggest to move all the code and add this adjustment to the correct date, from this function to `_withdrawTokens` to avoid potential misuse of `startFrom` parameter and add fuzzy-like tests passing a few random startFrom parameters as well as corner cases and compare with the expected result.

**Review by tjcloa — CHANGES_REQUESTED** [review]

**Comment by tjcloa** [comment]
missing deployment script that should deploy VestingLogic, register the new VestingLogic address in the VestingFactory contract and add VestingLogic contract hash to the Staking contact with `addContractCodeHash` function

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 16]
```diff
@@ -352,9 +358,22 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
...
-        vestingWhitelist[vesting] = true;
-        ITeamVesting(vesting).governanceWithdrawTokens(receiver);
-        vestingWhitelist[vesting] = false;
+        require(vestingRegistryLogic.isTeamVesting(vesting), "Only team vesting allowed");
+
+        ITeamVesting teamVesting = ITeamVesting(vesting);
+        uint256 teamVestingStartDate = teamVesting.startDate();
+        uint256 teamVestingCliff = teamVesting.cliff();
+
+        uint256 startFrom = teamVestingStartDate + teamVestingCliff;
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/63301217172621a0f99fa34d4787b5d25c42444e)

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 17]
```diff
@@ -139,9 +142,12 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
         }
 
         if (adjustedEnd < end) {
-            emit TeamVestingPartiallyCancelled(msg.sender, _receiver, adjustedEnd);
+            nextStartFrom = adjustedEnd + TWO_WEEKS;
+            emit TeamVestingPartiallyCancelled(msg.sender, _receiver, adjustedEnd, nextStartFrom);
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/63301217172621a0f99fa34d4787b5d25c42444e)

File: contracts/governance/Staking/modules/shared/CheckpointsShared.sol [hunk 18]
```diff
@@ -84,7 +84,8 @@ contract CheckpointsShared is StakingStorageShared, SafeMath96 {
     event TeamVestingPartiallyCancelled(
         address indexed caller,
         address receiver,
-        uint256 lastProcessedDate
+        uint256 lastProcessedDate,
+        uint256 nextStartFrom
     );
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/shared/CheckpointsShared.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/63301217172621a0f99fa34d4787b5d25c42444e)

File: contracts/governance/Vesting/ITeamVesting.sol [hunk 19]
```diff
@@ -16,6 +16,4 @@ interface ITeamVesting {
     function duration() external view returns (uint256);
 
     function tokenOwner() external view returns (address);
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Vesting/ITeamVesting.sol:18]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/63301217172621a0f99fa34d4787b5d25c42444e)

File: contracts/governance/Vesting/VestingLogic.sol (hunk 15 above)
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Vesting/VestingLogic.sol:137]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/63301217172621a0f99fa34d4787b5d25c42444e)

File: contracts/governance/Staking/modules/StakingWithdrawModule.sol [hunk 20]
```diff
@@ -352,9 +358,22 @@ contract StakingWithdrawModule is IFunctionsList, StakingShared, CheckpointsShar
...
+
+        uint256 startFrom = teamVestingStartDate + teamVestingCliff;
+        bool withdrawFlag = true;
+
+        uint256 nextStartFrom = startFrom;
+        bool notCompleted;
+
+        while (withdrawFlag) {
+            (nextStartFrom, notCompleted) = _cancelTeamVesting(vesting, receiver, nextStartFrom);
+            withdrawFlag = notCompleted ? true : false;
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Staking/modules/StakingWithdrawModule.sol:381]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/63301217172621a0f99fa34d4787b5d25c42444e)

**Review by tjcloa — COMMENTED** [review]
it used to be too expensive and confusing when Staking.governanceWithdrawVesting called 
 Vesting.governanceWithdrawTokens and then Vesting called back Staking.governanceWithdraw to do actual cancelling
1. @Cowsant 
where we need backwards compatibility
it was refactored at some point, but the function Vesting.governanceWithdrawTokens was meant to stay till BF contracts upgraded because BF team vestings are supposedly cancelled the old way - via vesting contracts, so this function was not removed for backwards interface compatibility for the sake of keeping interface because vesting contracts are not upgradeable and can be removed when we figure out what versions were deployed by BF and do Staking modules upgrade. 
so, pls revert it back for now: 


function governanceWithdrawTokens(address receiver) public {
        revert("operation not supported");
    } 

and also restore in ITeamVesting: function governanceWithdrawTokens(address receiver) external;

2. function StakingWithdrawModule.governanceWithdraw should be marked as deprecated in comments and further removed when we figure out BF

**Review by tjcloa — CHANGES_REQUESTED** [review]
it used to be too expensive and confusing when Staking.governanceWithdrawVesting called 
 Vesting.governanceWithdrawTokens and then Vesting called back Staking.governanceWithdraw to do actual cancelling.   
--------------------------------------------------------------------------------------------------------
Where we need backwards compatibility:   
1. pls revert it back

```
function governanceWithdrawTokens(address receiver) public {
        revert("operation not supported");
    } 
```

and also restore in ITeamVesting: `function governanceWithdrawTokens(address receiver) external`;  

2. function StakingWithdrawModule.governanceWithdraw should be marked as deprecated in comments and further removed when we figure out BF

File: contracts/governance/Vesting/VestingLogic.sol [hunk 21]
```diff
@@ -153,13 +170,13 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
         if (staking.allUnlocked() || isGovernance) {
```
[the remark this answers is not in the document]
Comment (tjcloa) [reply contracts/governance/Vesting/VestingLogic.sol]: why `if(staking.allUnlocked()` is removed? it should be able to process till the end, not till block.timestamp as mentioned in this comment.

File: contracts/governance/Vesting/VestingLogic.sol [hunk 22]
```diff
@@ -110,70 +117,73 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
...
-
-        /// @dev In the unlikely case that all tokens have been unlocked early,
-        ///   allow to withdraw all of them.
-        if (staking.allUnlocked() || isGovernance) {
-            end = endDate;
-        } else {
-            end = block.timestamp;
-        }
+        uint256 end = endAt < block.timestamp ? endAt : block.timestamp;
+        if (end > endDate && block.timestamp > endDate) end = endDate;
```
Comment (tjcloa) [inline contracts/governance/Vesting/VestingLogic.sol:172]: why if(staking.allUnlocked() is removed? it should be able to process till the end, not till block.timestamp as mentioned in this comment.

File: contracts/governance/Vesting/VestingLogic.sol [hunk 23]
```diff
@@ -110,70 +117,73 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
...
```
Comment (tjcloa) [inline contracts/governance/Vesting/VestingLogic.sol:171]:
```suggestion
        if (end > endDate) end = endDate;
```

File: deployment/deploy/2060-deploy-VestingLogic.js [hunk 24]
```diff
@@ -0,0 +1,85 @@
...
+        );
+        log(
+            col.bgBlue(
+                `>>> DONE. Requires Multisig (${multisigDeployment.address}) signatures to execute tx <<<`
+            )
+        );
+    } else if (hre.network.tags["mainnet"]) {
+        log(
+            col.bgBlue(
+                "Prepare and run SIP function in sips.js to create the proposal with params: getArgsSipSov625"
```
Comment (tjcloa) [inline deployment/deploy/2060-deploy-VestingLogic.js:79]: the sipArgs will be changed with the SIP number  
pls consider output of the sipArgs function data

File: hardhat/tasks/sips/args/sipArgs.js [hunk 25]
```diff
@@ -759,6 +759,27 @@ const getArgsSip0047 = async (hre) => {
     return { args, governor: "GovernorOwner" };
 };
 
+const getArgsSipSov625 = async (hre) => {
```
Comment (tjcloa) [inline hardhat/tasks/sips/args/sipArgs.js:762]: pls add onchain SIP creation and execution test

**Review by tjcloa — CHANGES_REQUESTED** [review]

File: deployment/deploy/2060-deploy-VestingLogic.js [hunk 26]
```diff
@@ -0,0 +1,85 @@
...
+        args: [vestingLogicDeployment.address],
+        log: true,
+        contract: "VestingFactory",
+    });
+
+    const vestingFactoryDeployment = await get("VestingFactory");
+    const vestingRegistry = await ethers.getContract("VestingRegistry");
+    const staking = await ethers.getContract("Staking");
+
+    /** VestingRegistry still owned by Exchequer for Mainnet & tetstnet */
```
Comment (tjcloa) [inline deployment/deploy/2060-deploy-VestingLogic.js:38]: pls consider differentiating upgrades based on actual ownership, not network and special cases

File: contracts/governance/Vesting/VestingLogic.sol (hunk 22 above)
Comment (cwsnt) [reply contracts/governance/Vesting/VestingLogic.sol]: I thought the initial condition was `if(staking.allUnlocked && isGovernance)` that's why i removed, apparently it OR condition.
Revert back the changes in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/eaf61228f6677387d6ee5b37b08ed83a61ceb269)

File: contracts/governance/Vesting/VestingLogic.sol (hunk 23 above)
Comment (cwsnt) [reply contracts/governance/Vesting/VestingLogic.sol]: Updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/eaf61228f6677387d6ee5b37b08ed83a61ceb269)

File: deployment/deploy/2060-deploy-VestingLogic.js (hunk 24 above)
Comment (cwsnt) [reply deployment/deploy/2060-deploy-VestingLogic.js:87]: Updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/eaf61228f6677387d6ee5b37b08ed83a61ceb269)

File: hardhat/tasks/sips/args/sipArgs.js (hunk 25 above)
Comment (cwsnt) [reply hardhat/tasks/sips/args/sipArgs.js:762]: Added in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/eaf61228f6677387d6ee5b37b08ed83a61ceb269)

File: deployment/deploy/2060-deploy-VestingLogic.js (hunk 26 above)
Comment (cwsnt) [reply deployment/deploy/2060-deploy-VestingLogic.js]: Apparently the VestingRegistry owned by governance, so i updated the deployment script in this Updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/eaf61228f6677387d6ee5b37b08ed83a61ceb269)

Comment (tjcloa) [reply deployment/deploy/2060-deploy-VestingLogic.js]: why isn't this one addressed?

Comment (tjcloa) [reply deployment/deploy/2060-deploy-VestingLogic.js]: actually, can be extreacted to a separate ticket - we need to refactor all the deployment scripts

Comment (tjcloa) [reply deployment/deploy/2060-deploy-VestingLogic.js]: resolved

**Review by tjcloa — APPROVED** [review]
pls resolve conflicts with the target branch
